principal/apis/eventstream/eventstream.go

```diff
@@ -189,6 +195,17 @@ func (s *Server) recvFunc(c *client, subs eventstreamapi.EventStream_SubscribeSe
 		return fmt.Errorf("panic: no recvq for agent %s", c.agentName)
 	}
 
+	if event.Target(incomingEvent) == event.TargetEventAck {
+		logCtx.Tracef("Received an ACK: resourceID %s eventID %s", event.ResourceID(incomingEvent), event.EventID(incomingEvent))
+		eventWriter, exists := s.eventWriters[c.agentName]
+		if !exists {
+			return fmt.Errorf("panic: event writer not found for agent %s", c.agentName)
```

Comment:
We don't expect eventWriter to be nil at this stage since we always initialize it when the agent invokes `Subscribe()`. If it is nil for some unexpected reason, we error and terminate the existing connection, forcing the agent to reconnect and thereby initializing a new eventWriter.  But this should never happen since we remove the eventWriter only when the stream is closed.